packages/strapi-design-system/src/DatePicker/DatePickerCalendar.tsx

```diff
@@ -54,98 +74,328 @@ export const DatePickerCalendarContent = ({
+    if (typeof year === 'number') {
+      /**
+       * This just to make TS happy, we're not going to get a
+       * number because we only use strings as options
+       */
+      return;
+    }
+
+    const updatedDate = localDate.set({ year: parseInt(year, 10) });
+    setLocalDate(updatedDate);
   };
 
-  const handleSelectDay = (date: Date) => () => {
+  const handleSelectDay = (date: DateValue) => () => {
     if (onChange) {
-      onChange(date);
+      onChange(date.toDate(TIME_ZONE));
     }
   };
 
+  const getDatesInWeek = makeGetDatesInWeek(startDate, locale);
+
   return (
-    <Box padding={4}>
-      <Box paddingBottom={4} paddingLeft={2} paddingRight={2}>
-        <Flex>
-          <SimpleMenu label={months[date.getMonth()]}>
-            {months.map((month) => (
-              <MenuItem key={month} onClick={() => handleMonthChange(month)}>
-                {month}
-              </MenuItem>
-            ))}
-          </SimpleMenu>
-          <Box paddingLeft={2}>
-            <SimpleMenu label={date.getFullYear()}>
-              {years.map((year) => (
-                <MenuItem key={year} onClick={() => handleYearChange(year)}>
-                  {year}
-                </MenuItem>
-              ))}
-            </SimpleMenu>
-          </Box>
-        </Flex>
-      </Box>
-      <RawTable colCount={7} rowCount={weeks.length + 1} initialCol={activeCol} initialRow={activeRow} role="grid">
-        <RawThead>
+    <Flex direction="column" alignItems="stretch" padding={4}>
+      <ToolbarFlex justifyContent="flex-start" paddingBottom={4} paddingLeft={2} paddingRight={2} gap={2}>
+        <SingleSelectInput
+          label={monthSelectLabel}
+          size="S"
+          value={months[localDate.month - 1]}
+          onChange={handleMonthChange}
+        >
+          {months.map((month) => (
+            <SingleSelectOption key={month} value={month}>
+              {month}
+            </SingleSelectOption>
+          ))}
+        </SingleSelectInput>
+        <SingleSelectInput
+          size="S"
+          value={localDate.year.toString()}
+          label={yearSelectLabel}
+          onChange={handleYearChange}
+        >
+          {years.map((year) => (
+            <SingleSelectOption key={year} value={year}>
+              {year}
+            </SingleSelectOption>
+          ))}
+        </SingleSelectInput>
+      </ToolbarFlex>
+      <RawTable colCount={7} rowCount={7} role="grid">
+        <RawThead aria-hidden>
           <RawTr aria-rowindex={1}>
-            <DatePickerTh>{sun}</DatePickerTh>
-            <DatePickerTh>{mon}</DatePickerTh>
-            <DatePickerTh>{tue}</DatePickerTh>
-            <DatePickerTh>{wed}</DatePickerTh>
-            <DatePickerTh>{thu}</DatePickerTh>
-            <DatePickerTh>{fri}</DatePickerTh>
-            <DatePickerTh>{sat}</DatePickerTh>
+            {weekDays.map((day) => (
+              <HeaderCell key={day}>{day}</HeaderCell>
```

Comment:
<img width="382" alt="Screenshot 2023-04-28 at 11 43 22" src="https://user-images.githubusercontent.com/2244375/235114320-2e924e90-f503-4399-a5b2-2e706c05d621.png">
<img width="382" alt="Screenshot 2023-04-28 at 11 43 18" src="https://user-images.githubusercontent.com/2244375/235114324-bf60776d-c2b4-42fc-b225-3ef81cfab5ad.png">


The clickable area/ focus outline of the Headercell is now much smaller than it used to be. Is it worth unifying with days?

Reply:
It shouldn't be focusable to be honest. It doesn't do anything and in examples like `react-aria` it's not, I think this improvement can be made in another PR this was just refactoring as opposing to improving the experience – wdyt?

Reply:
Works for me!